cp/packet_filters.c

```diff
@@ -766,11 +782,67 @@ parse_adc_rules(void)
 		/* Add Default rule */
 		adc_rule_id[rule_id - 1] = rule_id;
 		tmp_adc.rule_id = rule_id++;
+#ifdef MULTI_UPFS
+		RTE_SET_USED(dp_id);
+		struct adc_rules *adc = rte_zmalloc_socket(NULL, sizeof(struct adc_rules),
+							   RTE_CACHE_LINE_SIZE, rte_socket_id());
+		if (adc == NULL) {
+			rte_exit(EXIT_FAILURE, "Failure to allocate dedicated adc rule "
+				 "structure: %s (%s:%d)\n",
+				 rte_strerror(rte_errno),
+				 __FILE__,
+				 __LINE__);
+			return;
+		}
+
+		memcpy(adc, &tmp_adc, sizeof(tmp_adc));
+		adc_table[tmp_adc.rule_id] = adc;
+#else
 		if (adc_entry_add(dp_id, tmp_adc) < 0)
 			rte_exit(EXIT_FAILURE, "ADC entry add fail !!!");
+#endif /* MULTI_UPFS */
 		print_adc_rule(tmp_adc);
 
 	}
 	num_adc_rules = rule_id - 1;
 
 }
+#if defined(CP_BUILD) && defined(MULTI_UPFS)
+void
+init_pkt_filter_for_dp(void)
+{
+	uint32_t i;
+	struct dp_id dp_id = { .id = DPN_ID };
+
+	/* send pcc entries first */
+	for (i = 0; i < PCC_TABLE_SIZE; i++) {
+		if (pcc_filters[i] != NULL) {
+			if (pcc_entry_add(dp_id, *pcc_filters[i]) < 0)
+				rte_exit(EXIT_FAILURE, "PCC entry add fail!!!\n");
+		}
+	}
+
+	sleep(1);
```

Comment:
May be add comment. Why sleep is required.

Reply:
Good point. This was done because of the comment mentioned over [here](https://github.com/omec-project/ngic-rtc/blob/2ec6582d9561e393daff5556c556dbbc2bc08c77/cp/packet_filters.c#L652). Let me remove this line and see whether the dataplane still works fine.

Reply:
Okay. I tried removing this `sleep()` function and I did not see any DP crash (as the comment suggested). I tried running the `master` version without the `sleep()` function and it did not crash as well. This may mean that, in my testbed, I may be unable to recreate reported crash. I may need verification from someone else before I remove this suspiciously redundant line.

Reply:
Yeah. I believe you can keep it for now.

Reply:
Sorry. I missed this comment before. I am not sure why the `sleep()` is really needed. This `sleep()` was still there even before the ACK logic was added (in the master branch). Let me contact others (Somnath etc.) and get back to you on this.